tests/test_config.py

```diff
@@ -166,8 +188,10 @@ def test_ssl_config_combined(tls_certificate_pem_path):
     assert config.is_ssl is True
 
 
-def asgi2_app(scope):
-    async def asgi(receive, send):  # pragma: nocover
+def asgi2_app(scope: Scope) -> Callable:
```

Comment:
Could we use the `asgiref` types instead?

```py
def asgi2_app(scope: Scope) -> ASGI2Application:
    async def asgi(
        receive: ASGIReceiveCallable, send: ASGISendCallable
    ) -> None:  # pragma: nocover
        pass

    return asgi  # pragma: nocover

```

Reply:
That doesn't work because the `ASGI2Application` is _all of it_. That part is only the `callable` that I've annotated.

Reply:
> That doesn't work because the `ASGI2Application` is _all of it_. That part is only the `callable` that I've annotated.

I see. This is related to https://github.com/encode/uvicorn/pull/1067#discussion_r648638750 - if we don't want `Config(app)` to accept a callable, we might have to refactor these test methods.